test/brain_test.coffee

```diff
@@ -0,0 +1,253 @@
+
+  afterEach ->
+    @clock.restore()
+
+  describe 'Unit Tests', ->
+    describe '#mergeData', ->
+      it 'performs a proper merge with the new data taking precedent', ->
+        @brain.data =
+          1: 'old'
+          2: 'old'
+
+        @brain.mergeData
+          2: 'new'
+
+        expect(@brain.data).to.deep.equal
+          1: 'old'
+          2: 'new'
+
+       it 'emits a loaded event with the new data', ->
+         sinon.spy @brain, 'emit'
+         @brain.mergeData {}
+         expect(@brain.emit).to.have.been.calledWith('loaded', @brain.data)
+
+    describe '#save', ->
+      it 'emits a save event', ->
+        sinon.spy @brain, 'emit'
+        @brain.save()
+        expect(@brain.emit).to.have.been.calledWith('save', @brain.data)
+
+    describe '#resetSaveInterval', ->
+      it 'updates the auto-save interval', ->
+        sinon.spy @brain, 'save'
+        # default is 5s
+        @brain.resetSaveInterval 10
+        # make sure autosave is on
+        @brain.setAutoSave true
+
+        @clock.tick(5000)
+        # old interval has passed
+        expect(@brain.save).to.not.have.been.called
+        @clock.tick(5000)
+        # new interval has passed
+        expect(@brain.save).to.have.been.calledOnce
+
+    describe '#close', ->
+      it 'saves', ->
+        sinon.spy @brain, 'save'
+        @brain.close()
+        expect(@brain.save).to.have.been.calledOnce
+
+      it 'emits a close event', ->
+        sinon.spy @brain, 'emit'
+        @brain.close()
+        expect(@brain.emit).to.have.been.calledWith('close')
+
+      it 'saves before emitting the close event', ->
+        sinon.spy(@brain, 'save')
+        sinon.spy(@brain, 'emit').withArgs('close')
+        @brain.close()
+        expect(@brain.save).to.have.been.calledBefore(@brain.emit)
+
+      it 'stops auto-saving', ->
+        # make sure autosave is on
+        @brain.setAutoSave true
+        @brain.close()
+
+        # set up the spy after because 'close' calls 'save'
+        sinon.spy @brain, 'save'
+
+        @clock.tick(2*5000)
+        expect(@brain.save).to.not.have.been.called
+
+    describe '#get', ->
+      it 'returns the saved value', ->
+        @brain.data._private['test-key'] = 'value'
+        expect(@brain.get('test-key')).to.equal('value')
+
+      it 'returns null if object is not found', ->
+        expect(@brain.get('not a real key')).to.be.null
+
+    describe '#set', ->
+      it 'saves the value', ->
+        @brain.set('test-key', 'value')
+        expect(@brain.data._private['test-key']).to.equal('value')
+
+      it 'sets multiple keys at once if an object is provided', ->
+        @brain.data._private =
+          key1: 'val1'
+          key2: 'val1'
+
+        @brain.set
+          key2: 'val2'
+          key3: 'val2'
+
+        expect(@brain.data._private).to.deep.equal
+          key1: 'val1'
+          key2: 'val2'
+          key3: 'val2'
+
+      # Unable to understand why this behavior is needed, but adding a test
```

Comment:
Best I can tell, that is what `mergeData` is for. I did some spot checking around and every Brain persistence implementation I can find uses `mergeData` to inject data instead of using `set`.

Also, you can't initialize defaults on `loaded` because calling `set` from `loaded` triggers another `loaded` event (infinite loop). This behavior causes *every* `set` to emit `loaded`, which is confusing and, I'm pretty sure, wrong.